stravalib/model.py

```diff
@@ -432,379 +553,165 @@ class ActivityKudos(Athlete):
 
 
-class Split(BaseEntity):
+class Split(Split, BackwardCompatibilityMixin, DeprecatedSerializableMixin):
     """
     A split -- may be metric or standard units (which has no bearing
     on the units used in this object, just the binning of values).
     """
 
-    distance = Attribute(float, units=uh.meters)  #: Distance for this split
-    elapsed_time = (
-        TimeIntervalAttribute()
-    )  #: :class:`datetime.timedelta` of elapsed time for split
-    elevation_difference = Attribute(
-        float, units=uh.meters
-    )  #: Elevation difference for split
-    moving_time = (
-        TimeIntervalAttribute()
-    )  #: :class:`datetime.timedelta` of moving time for split
-    average_heartrate = Attribute(float)  #: Average HR for split
-    split = Attribute(int)  #: Which split number
-    pace_zone = Attribute(int)  #: (undocumented)
-    average_speed = Attribute(float, units=uh.meters_per_second)
-    average_grade_adjusted_speed = Attribute(float, units=uh.meters_per_second)
+    _unit_registry = {
+        'distance': 'meters',
+        'elevation_difference': 'meters',
+        'average_speed': 'm/s'
+    }
 
-    def __repr__(self):
-        return "<Split split={} distance={} elapsed_time={}>".format(
-            self.split, self.distance, self.elapsed_time
-        )
+    _field_conversions = {'elapsed_time': time_interval, 'moving_time': time_interval}
 
 
-class SegmentExplorerResult(LoadableEntity):
+class SegmentExplorerResult(
+    ExplorerSegment, BackwardCompatibilityMixin, DeprecatedSerializableMixin, BoundClientEntity
+):
     """
     Represents a segment result from the segment explorer feature.
 
     (These are not full segment objects, but the segment object can be fetched
     via the 'segment' property of this object.)
     """
 
-    _segment = None
-    id = Attribute(int)  #: ID of the segment.
-    name = Attribute(str)  #: Name of the segment
-    climb_category = Attribute(
-        int
-    )  #: Climb category for the segment (0 is higher)
-    climb_category_desc = Attribute(str)  #: Climb category text
-    avg_grade = Attribute(float)  #: Average grade for segment.
-    start_latlng = LocationAttribute()  #: Start lat/lon for segment
-    end_latlng = LocationAttribute()  #: End lat/lon for segment
-    elev_difference = Attribute(
-        float, units=uh.meters
-    )  #: Total elevation difference over segment.
-    distance = Attribute(float, units=uh.meters)  #: Distance of segment.
-    points = Attribute(str)  #: Encoded Google polyline of points in segment
-    starred = Attribute(
-        bool
-    )  #: Whether this segment is starred by authenticated athlete
+    _unit_registry = {
+        'elev_difference', 'meters',
+        'distance', 'meters'
+    }
 
-    @property
+    _latlng_extensions = extend_types('start_latlng', 'end_latlng', model_class='LatLon')
+
+    @lazy_property
     def segment(self):
         """Associated (full) :class:`stravalib.model.Segment` object."""
```

Comment:
so in the old API, there was a check for whether the ID existed. But here that is removed. and i see it's a lazily loaded element. is the ID being defined somewhere else now? i haven't play with the segment API endpoint yet so I will.

Reply:
To prevent code repetition, the checks for self.id and self.bound_client have moved to this `@lazy_property` decorator (you can see its full definition in `model.py`):

```python
            if obj.bound_client is None:
                raise exc.UnboundEntity(
                    f"Unable to fetch objects for unbound {obj.__class__} entity."
                )
            if obj.id is None:
                LOGGER.warning(f'Cannot retrieve {obj.__class__}.{fn.__name__}, self.id is None')
                return None
```